### DISCUSSION TITLE
How to bold the text in the picture?

### DISCUSSION THREAD

**Discussion by naohwen** [ImageMagick/ImageMagick#5519] [Help]
I want to print a PDF file, but the text in the file is too thin and it is not clear when printed. How can I bold the text?
![1662107292385](https://user-images.githubusercontent.com/82444429/188098121-c268bfe4-7312-4759-8cf8-a63d93268c22.png)

![image](https://user-images.githubusercontent.com/82444429/188097922-a63d2b46-7882-43e5-af19-f1c0d48a312f.png)

**Comment by snibgo** [comment]
When we have dark-gray text on a light-gray background, "`-contrast-stretch`" can be used, assuming the text that should be black occupies about 5 % of the paper, and about 80% of the paper should be white.
```
magick grayText.png -contrast-stretch 5x80% x.png
```
![x](https://user-images.githubusercontent.com/33812368/188192323-bae5c4ee-eab2-4d8e-8a3e-24e27ce4376e.png)

**Reply by naohwen** [reply]
Thank you very much for your answer, why does the picture turn yellow after processing? I think it would be better if it could handle the original PDF file. Is there any way to make the text of the original PDF thicker?I looked online and they said that you can use photoshop's filter, Minimum, which can make the text thicker

**Comment by snibgo** [comment]
The input image is not graysale. Pixels have red values greater then green values, which are larger than blue values, so the pixels are slightly orange/yellow. The operation increases contrast, so it exaggerates the colours. There are many other possible methods, such a processing just the L channel of Lab.

The characters contain fine detail. If strokes are made thicker, that will obscure gaps between strokes.

In the PDF file, are the characters recorded as characters (ie vector), or as raster images? If they are vector, then I suggest rendering the PDF at a larger density.

**Reply by naohwen** [reply]
Thank you very much for your answer. I use gimp to process, use Gaussian blur, and then sharpen, the text becomes thicker, and I use the printer to print it out and it will be clear. I don't know how to do this with magick.
![fea26d7b841a27f1fcd59e083d23cd3](https://user-images.githubusercontent.com/82444429/188303983-ff598b77-a92d-4864-a855-ca453688229f.png)
![a8084d3db8b8a2c06a1ad32118f54e5](https://user-images.githubusercontent.com/82444429/188303990-0a35b2db-8e41-4783-80ee-89710679deb0.jpg)
This is the code I wrote from looking at the documentation, but the effect is not the same
magick leaf.jpg -gaussian-blur 0.6 -unsharp 26.531 x2.png
![image](https://user-images.githubusercontent.com/82444429/188304103-ef34d4b4-6fb4-4246-b8f1-a8218c972ef4.png)

**Comment by snibgo** [comment]
The light areas within characters are so small that any blur will damage them. In my opinion, blurring reduces legibility.

IM has many options for sharpening, which can increase legibility. For example:
```
magick grayText.png -contrast-stretch 5x80% -sharpen 0x3 x1.png
```
![x1](https://user-images.githubusercontent.com/33812368/188325506-d0b11870-3d18-4fd6-9756-da8ba272401a.png)
